adler32_p.h

```diff
@@ -74,4 +80,20 @@ static inline uint32_t adler32_len_16(uint32_t adler, const unsigned char *buf,
     return adler | (sum2 << 16);
 }
 
+static inline uint32_t adler32_len_64(uint32_t adler, const unsigned char *buf, size_t len, uint32_t sum2) {
+    while (len >= 16) {
+        len -= 16;
+        DO16(adler, sum2, buf);
+        buf += 16;
+    }
+    /* Process tail (len < 16).  */
+    while (len--) {
```

Comment:
Perhaps it is possible to do `return adler32_len_16(adler, buf, sum2);` here?

Reply:
If we drop support for NO_DiViDE, we can simplify these a lot as then we don't need to worry about CHOP().

Reply:
That might be a good idea for a separate PR.

Reply:
Problem with using adler32_len_16 is that when/if NO_DIVIDE is set, the result can overflow. Modulo 65521 is essentially high 16 bits multiplied by 15 and added to low 16 bits... The result will be less than 2*65521, so it can do just one substraction to get same result as real nodulo.

Reply:
yes, I believe those are equivalent. The only difference is that `adler32_len_16` does a 
```
if (adler >= BASE)
      adler -= BASE
```
in the end, while that snippet does a `MOD(adler)`. But since the maximum increment to `adler` in the function will be fairly small (the sum of each byte in the buf, so <= 64*255 ?) then in this case both operations will be effectively the same.

Reply:
Yeah I'm not certain either. It might be same rarity though -- I think at least in `read_buf` buf size is typically multiple of 1024 at least...